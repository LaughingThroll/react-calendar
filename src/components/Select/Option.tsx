import React from "react"

interface IOption extends React.OptionHTMLAttributes<HTMLOptionElement> {
  title: string
}

const Option: React.FC<IOption> = ({ title, ...rest }) => {
  return (
    <option className="select__option" {...rest}>
      {title}
    </option>
  )
}

export default Option
