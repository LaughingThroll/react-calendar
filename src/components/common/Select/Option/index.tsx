import React from 'react'
import styles from './option.module.scss'

export interface OptionProps extends React.OptionHTMLAttributes<HTMLOptionElement> {
  title: string
}

const Option: React.FC<OptionProps> = ({ title, ...rest }) => {
  const { select__option } = styles
  return (
    <option className={select__option} {...rest}>
      {title}
    </option>
  )
}

export default React.memo(Option)
