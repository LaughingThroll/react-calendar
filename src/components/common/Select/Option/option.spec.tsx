import React from 'react'
import { shallow } from 'enzyme'
import Option, { OptionProps } from './'

const setUp = (props: OptionProps) => shallow(<Option {...props} />)

describe('Option render', () => {
  it('default render', () => {
    const tree = setUp({ title: 'Option' })

    expect(tree).toMatchSnapshot()
    expect(tree.find('option')).toHaveLength(1)
    expect(tree.find('option').text().includes('Option')).toBe(true)
  })
})
