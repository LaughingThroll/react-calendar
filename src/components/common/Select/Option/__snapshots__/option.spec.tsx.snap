// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Option render default render 1`] = `
<option
  className="select__option"
>
  Option
</option>
`;
